includes/fraud-prevention/class-buyer-fingerprinting-service.php

```diff
@@ -0,0 +1,87 @@
+<?php
+/**
+ * Class Buyer_Fingerprinting_Service
+ *
+ * @package WCPay\Fraud_Prevention
+ */
+
+namespace WCPay\Fraud_Prevention;
+
+/**
+ * Class Buyer_Fingerprinting_Service
+ */
+class Buyer_Fingerprinting_Service {
+	/**
+	 * Singleton instance.
+	 *
+	 * @var Buyer_Fingerprinting_Service
+	 */
+	private static $instance;
+
+	/**
+	 * Fraud prevention service instance.
+	 *
+	 * @var Fraud_Prevention_Service
+	 */
+	private $fraud_prevention_service;
+
+	/**
+	 * Buyer_Fingerprinting_Service constructor.
+	 *
+	 * @param Fraud_Prevention_Service $fraud_prevention_service Fraud Prevention instance.
+	 */
+	public function __construct( Fraud_Prevention_Service $fraud_prevention_service ) {
+		$this->fraud_prevention_service = $fraud_prevention_service;
+	}
+
+	/**
+	 * Returns singleton instance.
+	 *
+	 * @param null $fraud_prevention_service Fraud Prevention Service instance.
+	 * @return Buyer_Fingerprinting_Service
+	 */
+	public static function get_instance( $fraud_prevention_service = null ): self {
+		if ( null === self::$instance ) {
+			self::$instance = new self( $fraud_prevention_service ?? Fraud_Prevention_Service::get_instance() );
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * Sets a instance to be used in request cycle.
+	 * Introduced primarily for supporting unit tests.
+	 *
+	 * @param Buyer_Fingerprinting_Service|null $instance Instance of self.
+	 */
+	public static function set_instance( self $instance = null ) {
```

Comment:
This service exposes two idempotent public methods. Is there an actual need to override it for testing?